src/foam/nanos/auth/SubjectAuthService.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'SubjectAuthService',
+  extends: 'foam.nanos.auth.ProxyAuthService',
+
+  documentation: 'A decorator to check auth against every user in subject user path',
+
+  javaImports: [
+    'foam.core.X',
+    'foam.nanos.auth.Subject',
+    'foam.nanos.auth.User',
+    'foam.util.Auth',
+    'java.util.Map',
+    'java.util.HashMap'
+  ],
+
+  methods: [
+    {
+      name: 'check',
+      javaCode: `
+        if ( getDelegate().check(x, permission) ) return true;
+
+        Map<Long, Boolean> seen = new HashMap<>();
+        for ( User user : ((Subject) x.get("subject")).getUserPath() ) {
```

Comment:
The first check will be the same as the check on line 27, and isn't added to seen(), so is redundant.

Reply:
wouldn't the context from the check on line 27 different from the one from the first check? e.g., context with real user as agent, user as user whereas the one from the first check is specific to one user.

Reply:
Sorry, I got the other way around. The check on line 26 above would be the same as the last value from getUserPath().

Reply:
Could you explain how that is the case? I can see that the last one added to user path will be the user of the context used on line 26, but other than that can we guarantee realuser and group will be the same?

Reply:
I think we're saying the same thing, that last one added and the user in the context are the same.

Reply:
the last one added used be checked using the user itself, whereas the check on line 26 will use the context to do the check which could give us different result from the other one. Am I missing something here?